Ajuste do SPA, criação de páginas _Sample_ e adição de componentes

Guui-t opened the pull request:
## Implementado em #11 
- Adicionado o componente `BodyContent` (Agora é possível criar as páginas sem ter que definir _header_ e _sidebar_ para todas as páginas.

- Primeira implementação da página inicial (Adicionado somente o título e a barra de pesquisa

## O que foi adicionado

- Alterada a construção do código no arquivo __app.tsx_ . O `BodyContent` foi implementado como _parent_ de _components_ 
> Isso faz com que apenas o conteúdo da página seja atualizado, ao invés de toda a página junto com o `Header` e `Sidebar`. Isso deve arrumar o bug que fazia com que a `Sidebar` ficasse em um estado que não deveria.
Isso também permite que só seja criado o conteúdo das páginas ao adicionar uma nova página, não sendo necessário configurar o `Header` e `Sidebar` toda hora.

- Ajustado o tempo do _timer_ para a expansão da `Sidebar`
> Foi de `300` para `1000`

- Alterado o ícone do link para _Gerenciar biblioteca_

- Adicionada pequena transição ao trocar o _ActiveLink_
- Adicionados os componentes `HeadTitle`, `HomeThead`, e `BookCard`

> Um Define o _heading_ onde fica o título da página e o outro cria o cabeçalho da tabela para a página inicial, respectivamente. o `BookCard` está só de modelo por enquanto.

- Criadas páginas _sample_ para todas as páginas

> Agora não ficará mais dando erro ao trocar para uma página que tá nos links da `Sidebar`

Guui-t committed:
Adição do componente `BodyContent`

- Adcionado o componente `BodyContent` (Agora é possível criar as páginas sem ter que definir _header_ e _sidebar_ para todas as páginas.

- Primeira implementação da página inicial (Adcionado somente o titulo e a barra de pesquisa

Guui-t committed:
Ajuste do SPA, criação de páginas _Sample_ e adição de componentes

heber364 commented on src/components/BodyContent/index.tsx:
```diff
@@ -0,0 +1,48 @@
+import styles from "./Body.module.css";
+
+interface Props {
+    children: any;
```
Tipar children como Reactnode, importar do react.

heber364 commented on src/components/BodyContent/index.tsx:
```diff
@@ -0,0 +1,48 @@
+                </Box>
+            </div>
+        </div>
+    );
```
Usar componentes do próprio Chakra Ui.

heber364 commented on src/pages/user-control.tsx:
```diff
@@ -0,0 +1,12 @@
+    return <HeadTitle title="Membros" />;
+};
+
+export default UserControl;
```
Qual a finalidade dessa página?

Guui-t replied:
É a página que vai ter os membros, só coloquei algo pra ter a rota da página, pra parar de ficar dando erro. Mas por enquanto não tem nada ainda.

heber364 replied:
Ah sim, saquei.

heber364 commented on src/components/BodyContent/Body.module.css:
```diff
@@ -0,0 +1,17 @@
+
+.content {
+    height: 100%;
+}
```
Não usar Sass, já que estamos usando o Chakra Ui.

heber364 commented on src/pages/_app.tsx:
```diff
@@ -4,13 +4,16 @@ import { ChakraProvider } from "@chakra-ui/react";
-                <Component {...pageProps} />
+                <BodyContent>
+                    <Component {...pageProps} />
+                </BodyContent>
```
Envolvendo toda as as páginas  com  o BodyContent faz com que a página de login também tenha. Dá pra fazer uma condição lá no componente para ele se comportar de outra forma se a página ativa for a de login.

Guui-t committed:
ChakraUI Ajustado e Header/Sidebar removido da página de login

Guui-t committed:
Adicionado animação e polimento da aparência do seletor de imagem

Guui-t committed:
Ajustado o layout no modo de edição dos livros

Para seguir o que foi mudado quando adcionado

Guui-t committed:
Adcionada a biblioteca _react-cssfx-loading_ e ajustada animação de loading

Agora o circulo fica animado no centro da _div_ enquanto a imagem ainda não carregou.

faakit commented on src/components/AddBookModal/index.tsx:
```diff
@@ -112,6 +127,10 @@ export function AddBookModal() {
                         <FormControl flex="2">
                             <FormLabel
+                                title="Adicone uma imagem"
+                                style={hoverStyle}
```
no chakra o estilo de hover pode ser aplicado com a propriedade  `_hover`

faakit replied:
https://chakra-ui.com/docs/styled-system/style-props

faakit commented on src/components/EditBookModal/index.tsx:
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable react/jsx-indent-props */
 /* eslint-disable react/jsx-indent */
 import { useState, FormEvent } from "react";
+import { CircularProgress } from "react-cssfx-loading";
```
O mesmo pode ser atingido com o chakra circular-progress

https://chakra-ui.com/docs/components/circular-progress/usage

faakit commented on src/components/Header/index.tsx:
```diff
@@ -2,6 +2,7 @@
 /* eslint-disable react/jsx-indent-props */
 /* eslint-disable react/jsx-indent */
 import { useRef } from "react";
+import { render } from "react-dom";
```
por que isso foi importado ?

faakit commented on src/components/Sidebar/SidebarNav.tsx:
```diff
@@ -28,15 +28,21 @@ export function SidebarNav({ size, isOpen }: SidebarNavProps) {
         md: false,
     });
 
+    let mouseOver: any = null;
```
Não tipe nada com any, mouseOver deverá ser do tipo boolean se ele só recebe `true` ou `false`

Guui-t committed:
Atualizada as utilzações do hover e loading para usarem o chakra; Entrada de código errada removida; dependencia de loading removida

Guui-t committed:
Adcionado transição no hover

heber364 commented on src/components/AddBookModal/index.tsx:
```diff
@@ -122,8 +129,20 @@ export function AddBookModal() {
+                                position="relative"
+                                cursor="pointer"
                             >
-                                <Image src={imageDisplay} layout="fill" />
```
Não sei o nível de impacto exatamente, mas usar o _image_ do _nextjs_ é muito vantajoso na questão de otimização da imagem. 

Dê uma olhada nesse vídeo aqui: _https://www.youtube.com/watch?v=m149Itlw1zA_

Guui-t replied:
Vou ver se eu consigo colocar pra imagem da camera ficar pequena e a imagem do livro cobrir o container todo. Com o chakra é tranquilo, mas o next n fica certo

heber364 replied:
Então, esse vídeo ai mostra como usar o _chakra image_ junto com o _next image_, a parte de otimização do nextjs e a estilização do chakra.

Guui-t replied:
Mesmo colocando desse jeito, ele n deixa a imagem do tamanho do container, ou a imagem de label fica muito grande ou a imagem que é adcionada fica muito pequena

heber364 commented on src/components/EditBookModal/index.tsx:
```diff
@@ -90,13 +90,33 @@ export function EditBookModal({ book, isOpen, onClose }: EditBookModalProps) {
                                 border="1px"
                                 borderColor="gray.200"
                             >
-                                <Image src={imageDisplay} layout="fill" />
```
A mesma questão do review comment anterior sobre _next image_.

Guui-t committed:
Primeira implementação da exibição dos livros na página inicial e pesquisa

Implementando a primeira tentativa de fazer um card em lista

Guui-t committed:
Corrigido sintaxe

Guui-t committed:
Adcionada propriedade para retirar erro

Guui-t committed:
adcionado requestBooksFirebase na home e search

Guui-t committed:
Ajustado tamanho das imagens no sidebar; Ajustado cover das imagens; Adcionado botão de ScrollTop

faakit commented on src/components/BooksDisplay/index.tsx:
```diff
@@ -0,0 +1,210 @@
+                                {new Date(
+                                    new Date().setDate(
+                                        // eslint-disable-next-line prettier/prettier
+                                        new Date(book.createdAt).getDate() + 7)
```
por que o construtor de data é chamado 3 vezes? isso pode ser atingido com apenas 1 chamada

heber364 commented on src/components/HomeThead/index.tsx:
```diff
@@ -0,0 +1,41 @@
+    hasHead: boolean;
+}
+
+export function TableLibraryDisplay({
```
A pasta do componente seria bom ter o mesmo nome da _function_ do arquivo do index.tsx. Ao invés de HomeThead,  seria TableLibraryDisplay.

heber364 commented on src/components/ScrollTopButton/index.tsx:
```diff
@@ -0,0 +1,55 @@
+    return (
+        <Button
+            title="Voltar ao topo"
+            style={{
```
Acho que as propriedades aqui não precisam ser envolvidas por esse _"style"_.

Guui-t committed:
corrigido estilização do botão, ajustado nome da pasta
